ApplicationLibCode/ProjectDataModel/WellLog/RimWellLogCalculatedCurve.cpp

```diff
@@ -0,0 +1,479 @@
+///
+//--------------------------------------------------------------------------------------------------
+void RimWellLogCalculatedCurve::connectWellLogCurveChangedToSlots( RimWellLogCurve* wellLogCurve )
+{
+    if ( !wellLogCurve ) return;
+
+    wellLogCurve->dataChanged.connect( this, &RimWellLogCalculatedCurve::onWellLogCurveChanged );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimWellLogCalculatedCurve::disconnectWellLogCurveChangedFromSlots( RimWellLogCurve* wellLogCurve )
+{
+    if ( !wellLogCurve ) return;
+
+    wellLogCurve->dataChanged.disconnect( this );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+double RimWellLogCalculatedCurve::calculateValue( double firstValue, double secondValue, Operators operatorValue )
+{
+    if ( operatorValue == Operators::ADD )
+    {
+        return firstValue + secondValue;
+    }
+    if ( operatorValue == Operators::SUBTRACT )
+    {
+        return firstValue - secondValue;
+    }
+    if ( operatorValue == Operators::MULTIPLY )
+    {
+        return firstValue * secondValue;
+    }
+    if ( operatorValue == Operators::DIVIDE )
+    {
+        return firstValue / secondValue;
+    }
+    return 0.0;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+std::vector<double> RimWellLogCalculatedCurve::depthValuesFromSource( RiaDefines::DepthTypeEnum depthType ) const
+{
+    if ( !m_firstWellLogCurve || !m_firstWellLogCurve->curveData() || !m_secondWellLogCurve || !m_secondWellLogCurve->curveData() )
+    {
+        return {};
+    }
+
+    if ( m_depthSource() == DepthSource::FIRST )
+    {
+        return m_firstWellLogCurve->curveData()->depths( depthType );
+    }
+    else if ( m_depthSource() == DepthSource::SECOND )
+    {
+        return m_secondWellLogCurve->curveData()->depths( depthType );
+    }
+    else if ( m_depthSource() == DepthSource::UNION )
+    {
+        return unionDepthValuesFromCurves( depthType );
+    }
+
+    return {};
+}
+
+//--------------------------------------------------------------------------------------------------
+/// As each depth value is repeated twice, the depth values are sorted, the duplicates cannot
+/// be removed with unionDepths.erease() and std::unique. Instead, the duplicates are removed
+/// manually.
+//--------------------------------------------------------------------------------------------------
+std::vector<double> RimWellLogCalculatedCurve::unionDepthValuesFromVectors( const std::vector<double>& firstDepths,
```

Comment:
Try to use a different approach here, maybe test if a value already is present in a map of doubles

Reply:
Done as discussed